src/components/manage/Blocks/ToC/variations/DefaultTocRenderer.jsx

```diff
@@ -9,14 +9,18 @@ import { map } from 'lodash';
 import { List } from 'semantic-ui-react';
 import { FormattedMessage, injectIntl } from 'react-intl';
 import AnchorLink from 'react-anchor-link-smooth-scroll';
+import Slugger from 'github-slugger';
 
 const RenderListItems = ({ items, data }) => {
   return map(items, (item) => {
-    const { id, level, title } = item;
+    const { id, level, title, override_toc, plaintext } = item;
+    const slug = override_toc
+      ? Slugger.slug(plaintext)
+      : Slugger.slug(title) || id;
     return (
       item && (
         <List.Item key={id} className={`item headline-${level}`} as="li">
-          <AnchorLink href={`#${id}`}>{title}</AnchorLink>
+          <AnchorLink href={`#${slug}`}>{title}</AnchorLink>
```

Comment:
I expected that clicking the link in the ToC would update the location to include the hash (so that it's easy to copy from the address bar), but it did not.

Reply:
[c33f634](https://github.com/plone/volto/pull/4287/commits/c33f634e32bcd7eb56ae80d36dd5258f2aeb101e)